plugins/synapticsmst/synapticsmst-device.c

```diff
@@ -504,32 +541,502 @@ synapticsmst_device_get_flash_checksum (SynapticsMSTDevice *device,
+			val = (guint8)(message[byte] ^ remainder);
+			remainder = CRC8_table[val];
+		}
+	}
+	else {
+		for ( byte = 0; byte < length; ++byte)	{
+			val = (guint8)(message[byte] ^ ( remainder>>8));
+			remainder = CRC16_table[val] ^ (remainder << 8);
+		}		
+	}  
+    return (remainder);
+}
+/*
+guint8 	synapticsmst_device_disable_flash_write_protect(SynapticsMSTDevice *device, guint8 mode)
+{
+	guint32 tmp[5];
+	guint8 rc;
+
+	g_autoptr(SynapticsMSTConnection) connection = NULL;
+	SynapticsMSTDevicePrivate *priv = GET_PRIVATE (device);
+	connection = synapticsmst_common_new (priv->fd, priv->layer, priv->rad);
+
+	rc = synapticsmst_common_rc_get_command (connection, UPDC_READ_FROM_MEMORY,
+							 (gint)((sizeof(tmp)/sizeof(tmp[0]))*4), (gint)0x2000c8, (guint8*)tmp);
+	if (rc) {
+		return FALSE;
+	}
+	if(mode == TRUE)
+		tmp[ 0 ] |= BIT( 18 );		// GPIO_TX
+	else
+		tmp[ 0 ] &= ~BIT( 18 );		// GPIO_TX
+	tmp[ 1 ] &= ~BIT( 18 );		// GPIO_OEN
+	tmp[ 3 ] &= ~BIT( 18 );		// GPIO_TX_SEL
+	tmp[ 4 ] &= ~BIT( 18 );		// GPIO_MCU_SEL
+	rc = synapticsmst_common_rc_set_command (connection, UPDC_WRITE_TO_MEMORY,
+							 (sizeof(tmp)/sizeof(tmp[0]))*4, 0x2000c8, tmp);
+	if (rc) {
+		return FALSE;
+	}
+	return TRUE;
+}
+*/
+guint8 
+synapticsmst_device_set_flash_sector_erase(SynapticsMSTDevice *device, guint16 rc_cmd, guint16 offset)
+{
+	guint16 usData;
+	g_autoptr(SynapticsMSTConnection) connection = NULL;
+	SynapticsMSTDevicePrivate *priv = GET_PRIVATE (device);
+	connection = synapticsmst_common_new (priv->fd, priv->layer, priv->rad);
+	/* Need to add Wp control ? */
+	//rc = synapticsmst_device_disable_flash_write_protect(device, TRUE);	 
+	usData = rc_cmd + offset;
+	return synapticsmst_common_rc_set_command (connection,
+						UPDC_FLASH_ERASE,
+						2, 0, (guint8 *)&usData);
+
+}
+
+guint8
+synapticsmst_device_update_ESM(SynapticsMSTDevice *device, guint8	*payload_data, GError **error)
+{
+	guint32 esmSize = ESM_CODE_SIZE;
+	guint32 checksum = 0;
+	guint32 flash_checksum = 0;
+	gboolean isESMupdted = FALSE;
+	gboolean  eraseFail = FALSE;
+	guint8 rc = 0xFF;
+	guint32 i = 0;
+	guint32 writeOffset, writeIndex;
+	guint32 write_loops = 0;
+	guint32 retryCount, unitSize;
+	SynapticsMSTDevicePrivate *priv = GET_PRIVATE (device);
+	g_autoptr(SynapticsMSTConnection) connection = NULL;
+	guint8 *ptrESMcode = &payload_data[EEPROM_ESM_OFFSET]; 
+
+	connection = synapticsmst_common_new (priv->fd, priv->layer, priv->rad);
+
+	checksum = 0;
+	for (i = 0; i < esmSize; i++) {
+		checksum += *(payload_data + EEPROM_ESM_OFFSET +i);
+	}
+	flash_checksum = 0;
+	if (synapticsmst_device_get_flash_checksum (device,
+				  esmSize, EEPROM_ESM_OFFSET, &flash_checksum, error)) 
+	{
+		if (checksum != flash_checksum) 
+		{
+			/* erase ESM firmware */	
+			for (i=0; i<4; i++)
+			{				
+				rc = synapticsmst_device_set_flash_sector_erase(device, FLASH_SECTOR_ERASE_64K, i + 4);
+				if (rc)
+				{
+					g_set_error_literal (error,	G_IO_ERROR,	G_IO_ERROR_INVALID_DATA,
```

Comment:
Funny whitespace on this, I would newline after every comma and then align the start of each line together (which matches lots of cases of `g_set_error_literal` across the project.

Reply:
Some of then is aligned with my IDE (vscode), not sure why became un-aligned.

Reply:
In this case they're all combined into one line.  It might have been from a bad copy/paste result or an vscode extension if you installed one.

It's been on my TODO for a while to figure out some best practices with vscode and document them in the style document or something that vscode can import for settings so this doesn't need to be such a manual process.